src/subcommand/server.rs

```diff
@@ -1287,11 +1287,14 @@ impl Server {
         };
       };
 
-      if let Some(delegate) = inscription.delegate() {
-        inscription = index
-          .get_inscription_by_id(delegate)?
-          .ok_or_not_found(|| format!("delegate {inscription_id}"))?
-      }
+      // return the first inscription where the delegate exists
+      // probably worth considering some sort of limitation to mitigate "DoSability"
+      let inscription_override = inscription
+        .delegates()
+        .iter()
+        .find_map(|delegate| index.get_inscription_by_id(*delegate).unwrap_or(None));
```

Comment:
Can we add a length number to avoid the Dos concern?
And extract a function to remove the repeat code below.
```rust
fn find_inscription<'a>(delegates: &'a [usize], index: &'a Index, max_length: usize) -> Option<&'a Inscription> {
    delegates
        .iter()
        .take(max_length)
        .find_map(|delegate| index.get_inscription_by_id(*delegate).unwrap_or(None))
}

// Usage
let inscription_override = find_inscription(inscription.delegates(), &index, max_length);
```